pkg/container-collection/container-collection.go

```diff
@@ -24,28 +24,93 @@ package containercollection
 import (
 	"sync"
 
+	dockerclient "github.com/docker/docker/client"
+	"k8s.io/client-go/kubernetes"
+
 	pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
 	"github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/pubsub"
+	"github.com/kinvolk/inspektor-gadget/pkg/runcfanotify"
 )
 
 // ContainerCollection holds a set of containers. It can be embedded as an
 // anonymous struct to help other structs implement the ContainerResolver
 // interface. For this reason, some methods are namespaced with 'Container' to
 // make this clear.
 type ContainerCollection struct {
-	// Keys:   Container ID
-	// Values: pb.ContainerDefinition
+	// Keys:   containerID string
+	// Values: container   *pb.ContainerDefinition
 	containers sync.Map
 
 	// subs contains a list of subscribers of container events
 	pubsub *pubsub.GadgetPubSub
+
+	// containerEnrichers are functions that automatically add metadata
+	// upon AddContainer. The functions return true on success or false if
+	// the container is meant to be dropped.
+	containerEnrichers []func(container *pb.ContainerDefinition, initial bool) (ok bool)
+
+	// dockercli is the Docker API client to be used by
+	// WithDockerEnrichment
+	dockercli *dockerclient.Client
+
+	// client to talk to the k8s API server to get information about pods
+	k8sClient *kubernetes.Clientset
+
+	// runcNotifier to be used by WithRuncFanotify
+	runcNotifier *runcfanotify.RuncNotifier
```

Comment:
I think we don't need to make these members. Couldn't they be declared in the scope of the different `ContainerCollectionOption` implementations?

Reply:
Removed